### PR TITLE
feat: update support for passthrough rollouts

### PR DESCRIPTION
**Pull request by suthar26**
- add new passthrough logic for rollout checking

### CODE REVIEW

**Comment by suthar26** [comment]
* **#257** <a href="https://app.graphite.dev/github/pr/DevCycleHQ/go-server-sdk/257?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @suthar26 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: api/model_public_project.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ type Project struct {
 type ProjectSettings struct {
 	EdgeDB EdgeDBSettings `json:"edgeDB"`
 	OptIn  OptInSettings  `json:"optIn"`
+	DisablePassthroughRollouts bool `json:"disablePassthroughRollouts" default:"false"`
```
Comment (JamieSinn) [inline api/model_public_project.go:13]: defaults aren't used this way

Comment (JamieSinn) [reply api/model_public_project.go:13]:
```suggestion
	DisablePassthroughRollouts bool `json:"disablePassthroughRollouts"` 
```

File: bucketing/bucketing_test.go [hunk 2]
```diff
@@ -53,7 +54,15 @@ func TestBucketing_RolloutGatesUser(t *testing.T) {
 		Email:  "test@email.com",
 	}.GetPopulatedUser(&api.PlatformData{})
 
-	config, err := newConfig(test_config, "", "", "")
+	var test_config_json map[string]interface{}
+	err := json.Unmarshal( test_config, &test_config_json )
```
Comment (JamieSinn) [inline bucketing/bucketing_test.go:58]: run `go fmt`

File: bucketing/testdata/fixture_test_config.json [hunk 3]
```diff
@@ -21,10 +21,7 @@
             "type": "user",
             "subType": "email",
             "comparator": "=",
-            "values": [
-              "test@email.com",
-              "test2@email.com"
-            ]
+            "values": ["test@email.com", "test2@email.com"]
```
Comment (JamieSinn) [inline bucketing/testdata/fixture_test_config.json:24]: this should be in the most expanded format possible

File: bucketing/testdata/fixture_test_config.json
Comment (JamieSinn) [inline bucketing/testdata/fixture_test_config.json:1]: Make sure all the other test config files are updated to pass this change too.

**Review by JamieSinn — CHANGES_REQUESTED** [review]

File: bucketing/bucketing_test.go [hunk 4]
```diff
@@ -53,7 +54,15 @@ func TestBucketing_RolloutGatesUser(t *testing.T) {
 		Email:  "test@email.com",
 	}.GetPopulatedUser(&api.PlatformData{})
 
-	config, err := newConfig(test_config, "", "", "")
+	var test_config_json map[string]interface{}
+	err := json.Unmarshal(test_config, &test_config_json)
+	require.NoError(t, err)
+	test_config_json["project"].(map[string]interface{})["settings"].(map[string]interface{})["disablePassthroughRollouts"] = true
```
Comment (JamieSinn) [inline bucketing/bucketing_test.go:60]: This should be a separate config that isn't hardcoding the path/value.

Comment (JamieSinn) [reply bucketing/bucketing_test.go:60]: or better yet - update all the configs to have a valid setting and ensure that it doesn't break existing tests

**Review by JamieSinn — APPROVED** [review]

**Review by jsalaber — COMMENTED** [review]
LGTM, but will wait approval from jamie

**Review by jsalaber — APPROVED** [review]

**Pull request merged**